lib/packwerk/checker.rb

```diff
@@ -0,0 +1,20 @@
+# typed: true
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "packwerk/reference_lister"
+
+module Packwerk
+  module Checker
```

Comment:
This interface is a good idea, and `Checker` looks like a good name based on my current understanding of our naming conventions 👍